docs(docs-infra): renders inline code snippets in docs pills

SkyZeroZx opened the pull request:
Improves the rendering of `DocsPillRow` to correctly display inline code snippets (using backticks) within the pill labels.


## What is the current behavior?
 https://angular.dev/api/core/createComponent
 
<img width="380" height="493" alt="image" src="https://github.com/user-attachments/assets/fa0c873b-a336-402b-817c-f6c4e986522e" />

## What is the new behavior?
<img width="392" height="364" alt="image" src="https://github.com/user-attachments/assets/2f5746fc-ac66-486d-a04e-5fe60c58e157" />


## Other information

I'm not sure this is the best approach, but it is more testable since it doesn't require parsing the `<code>` tag in `docs-pill-row.tsx`. I believe it wouldn't be insecure, considering that the API is generated at build time.

SkyZeroZx commented on adev/shared-docs/pipeline/api-gen/rendering/transforms/jsdoc-transforms.mts:
```diff
@@ -165,6 +165,15 @@ function getHtmlAdditionalLinks<T extends HasJsDocTags>(entry: T): LinkEntryRend
 
+/**
+ * Converts inline markdown syntax (like backticks for code) to HTML in link labels.
+ * This handles code blocks within link text, e.g., `ViewContainerRef.createComponent`
```
Done

JeanMeche commented:
Can we also improve the spacing around the code blocks: 

https://ng-dev-previews-fw--pr-angular-angular-65562-adev-prev-i2jg70ek.web.app/api/core/createComponent#usage-notes

SkyZeroZx commented:
> Can we also improve the spacing around the code blocks:
> 
> https://ng-dev-previews-fw--pr-angular-angular-65562-adev-prev-i2jg70ek.web.app/api/core/createComponent#usage-notes

Are you referring to the space between the the code and next comment ?
<img width="843" height="806" alt="image" src="https://github.com/user-attachments/assets/5ee35434-5a2b-472c-aadb-1a209af4e265" />

SkyZeroZx committed:
docs(docs-infra): renders inline code snippets in docs pills

Improves the rendering of `DocsPillRow` to correctly display inline code snippets (using backticks) within the pill labels.
